### PR TITLE
mutable-content support

### PR DESCRIPTION
**Pull request by Necsord**
Per https://developer.apple.com/reference/usernotifications/unnotificationserviceextension

Apple introduced new field `mutable-content` with the release of iOS 10. This field is designated for iOS notification extensions that allow download of additional information.

### CODE REVIEW

**Comment by fjalvarezdd** [comment]
Hello,

Thanks for this PR. I need this piece of code to be merged asap. Does anyone reviewed it ? Thanks in advance

**Comment by seyfer** [comment]
@Ocramius, please. It's really needed. I don't see issues here, could be merged.

File: library/ZendService/Apple/Apns/Message.php [hunk 1]
```diff
@@ -55,6 +55,12 @@ class Message
     protected $sound;
 
     /**
+     * Mutable Content
+     * @var int|null
+     */
+    protected $mutableContent;
```
Comment (Ocramius) [inline library/ZendService/Apple/Apns/Message.php]: `private`

File: library/ZendService/Apple/Apns/Message.php [hunk 2]
```diff
@@ -256,6 +262,32 @@ public function setSound($sound)
     }
 
     /**
+     * Get Mutable Content
+     * 
+     * @return int|null
+     */
+    public function getMutableContent()
```
Comment (Ocramius) [inline library/ZendService/Apple/Apns/Message.php]: Not needed

File: library/ZendService/Apple/Apns/Message.php [hunk 3]
```diff
@@ -256,6 +262,32 @@ public function setSound($sound)
...
+    /**
+     * Set Mutable Content
+     * 
+     * @param int|null $value
+     * @returns Message
+     */
+    public function setMutableContent($value)
+    {
+        if ($value !== null && !is_int($value)) {
+            throw new Exception\InvalidArgumentException('Mutable Content must be null or an integer');
```
Comment (Ocramius) [inline library/ZendService/Apple/Apns/Message.php]: Add the type of what was given to the exception message

File: tests/ZendService/Apple/Apns/MessageTest.php [hunk 4]
```diff
@@ -127,6 +127,27 @@ public function testSetSoundThrowsExceptionOnNonString()
         $this->message->setSound(12345);
     }
 
+    public function testSetMutableContentThrowsExceptionOnNonInteger()
+    {
+        $this->setExpectedException('InvalidArgumentException');
+        $this->message->setMutableContent("string");
```
Comment (Ocramius) [inline tests/ZendService/Apple/Apns/MessageTest.php]: Multiple values need to be tested

File: library/ZendService/Apple/Apns/Message.php [hunk 5]
```diff
@@ -256,6 +262,32 @@ public function setSound($sound)
...
+     */
+    public function getMutableContent()
+    {
+        return $this->mutableContent;
+    }
+
+    /**
+     * Set Mutable Content
+     * 
+     * @param int|null $value
```
Comment (Ocramius) [inline library/ZendService/Apple/Apns/Message.php]: Is any integer acceptable? What about negative values? Is there documentation for this?

**Comment by Ocramius** [comment]
@seyfer this library is not actively maintained, so it may take ages for anything to go through, sorry.

**Comment by seyfer** [comment]
@Ocramius, is it possible to become a maintainer? Now I'm one of the maintainers here
https://github.com/Ph3nol/NotificationPusher

And I have moved this library forward. But it's APNS implementation depends on **ZendService_Apple_Apns**
So, it's in my interest to move forward things here also.

File: library/ZendService/Apple/Apns/Message.php (hunk 5 above)
Comment (Necsord) [inline library/ZendService/Apple/Apns/Message.php]: I could not find anything specific, documentations suggest setting it to `1`, so I would assume this can change in future to other values

Comment (Ocramius) [inline library/ZendService/Apple/Apns/Message.php]: @Necsord then we should currently validate the value to be in the range of `[null, 1]`, and reject other values

File: tests/ZendService/Apple/Apns/MessageTest.php [hunk 6]
```diff
@@ -127,6 +127,59 @@ public function testSetSoundThrowsExceptionOnNonString()
...
+    }
+
+    /**
+     * @return array
+     */
+    public function provideSetMutableContentThrowsExceptionOnNonIntegerOneOrNullData()
+    {
+        return array(
+            'unsupported integer value' => array(
+                'value' => 2,
```
Comment (Ocramius) [inline tests/ZendService/Apple/Apns/MessageTest.php]: Add also `0` and `-1`, which are edge-case values

**Comment by Ocramius** [comment]
LGTM :+1: 

I see that PHP 5.3 no longer runs on travis - that's an unrelated issue

**Comment by weierophinney** [comment]
Thanks, @Necsord; merged to develop for release with 1.3.0.
